client/services/SolutionHandlerServiceSpec.js

```diff
@@ -0,0 +1,227 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for SolutionHandlerService.
+ */
+
+describe('SolutionHandlerService', function() {
+  var SolutionHandlerService;
+  var TaskObjectFactory;
+  var orderedTasks;
+  var auxiliaryCode;
+
+  beforeEach(module('tie'));
+  beforeEach(inject(function($injector) {
+    SolutionHandlerService = $injector.get(
+      'SolutionHandlerService');
+  }));
+
+  // Mock tasks for preprocessing.
+  beforeEach(inject(function($injector) {
+    TaskObjectFactory = $injector.get(
+      'TaskObjectFactory');
+
+    var taskDict = [
+      {
+        instructions: [''],
+        prerequisiteSkills: [''],
+        acquiredSkills: [''],
+        inputFunctionName: null,
+        outputFunctionName: null,
+        mainFunctionName: 'mockMainFunction',
+        correctnessTests: [{
+          input: 'testcase_0',
+          allowedOutputs: [true]
+        }, {
+          input: 'testcase_1',
+          allowedOutputs: [true]
+        }],
+        buggyOutputTests: [{
+          buggyFunctionName: 'AuxiliaryCode.mockAuxiliaryCodeOne',
+          messages: [
+            "Mock BuggyOutputTest Message One for task1",
+            "Mock BuggyOutputTest Message Two for task1",
+            "Mock BuggyOutputTest Message Three for task1"
+          ]
+        }],
+        performanceTests: []
+      },
+      {
+        instructions: [''],
+        prerequisiteSkills: [''],
+        acquiredSkills: [''],
+        inputFunctionName: null,
+        outputFunctionName: null,
+        mainFunctionName: 'mockMainFunction',
+        correctnessTests: [{
+          input: 'testcase_2',
+          allowedOutputs: [false]
+        }, {
+          input: 'testcase_3',
+          allowedOutputs: [false]
+        }],
+        buggyOutputTests: [{
+          buggyFunctionName: 'AuxiliaryCode.mockAuxiliaryCodeTwo',
+          messages: [
+            "Mock BuggyOutputTest Message One for task2",
+            "Mock BuggyOutputTest Message Two for task2",
+            "Mock BuggyOutputTest Message Three for task2"
+          ]
+        }],
+        performanceTests: []
+      }
+    ];
+
+    orderedTasks = taskDict.map(function(task) {
+      return TaskObjectFactory.create(task);
+    });
+  }));
+
+  // Mock auxiliary code for processing.
+  beforeEach(function() {
+    auxiliaryCode = [
+      'class AuxiliaryCode(object):',
+      '    @classmethod',
+      '    def mockAuxiliaryCodeOne(cls, input):',
+      '        return True',
+      '    @classmethod',
+      '    def mockAuxiliaryCodeTwo(cls, input):',
+      '        return False'
+    ].join('\n');
+  });
+
+  describe('processSolutionAsync', function() {
+    describe('correctnessTests', function() {
+      it('should be passed on both task1 and task2', function() {
```

Comment:
should check both task1 and task2 to verify that the learner has the correct answer